src/include/msvc.h

```diff
@@ -117,6 +117,17 @@ WT_READ_BARRIER(void)
     _mm_lfence();
 }
 
+/*
+ * WT_READ_BARRIER_WEAK_MEMORDER --
+ *	MSVC implementation of WT_READ_BARRIER_WEAK_MEMORDER. Since we only support
+ *	x86 on windows and x86 has a strong memory model, we only need a compiler barrier here.
```

Comment:
I'd go the other way on this, and make it strong. Platform support is really a product level decision, and that could change without us remembering this choice and changing it.

Reply:
That will cause the regression still exist on windows on x86. I think what I can do is add a check here and if it's x86 use the weaker barrier and in other case use a strong barrier.

Reply:
done